src/Screen/FB/TopWindow.cpp

```diff
@@ -27,6 +27,7 @@ Copyright_License {
 #include "Event/Poll/Loop.hpp"
 #include "Event/Queue.hpp"
 #include "Event/Globals.hpp"
+#include "Interface.hpp"
```

Comment:
The "Screen" library should not include this header, because it's a higher-level collection of global variables (aka "shit code needs to live somewhere, and this header it is") - layering violation.
Maybe the cursor size setting should be copied to a `TopWindow` variable?

Reply:
Sorry, my mistake. Guess I'll have to check out the architecture. Also: the include should probably be tied to the pre-compiler conditional as well, i suppose.

Reply:
At that: Is it considered better style to have both includes and code in the same pre-compiler conditional or separate? (I'm looking at `USE_FB` in the same file, where both seem to be kept in the same conditional bracket)

Reply:
I don't understand the question.

The "architecture" is not documented, no need to be sorry.
The `Screen` library (the low-level UI toolkit which higher levels like `Form` build on top of) should be generic and not depend on XCSoar's specific libraries and global variables. If some global configuration needs to be known by the `Screen` library, it should be passed during initialization.
There are still some historic layering violations; after many years of code cleanup, it's far from perfect. But I'd like new code to be as good as possible, while still trying to improve the old code.

Reply:
Ok, I'll try my best (well, maybe I'm just a little confused here). Added a commit where the cursor size is an attribute of `TopWindow`. Since I would like to instantly apply the settings change (i.e. not require the user to restart), I updated the value in the `LayoutConfigPanel`. Also the update encompasses the more generic `DisplaySettings` as a whole. Let me know if this is not acceptable.